org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -3989,6 +3989,8 @@ private void completionOnSingleNameReference(ASTNode astNode, ASTNode astNodePar
 					}
 					System.arraycopy(keywords, 0, keywords = new char[count][], 0, count);
 					findKeywords(this.completionToken, keywords, false, false);
+					// complete types as it is done for version < J17
+					findTypesAndPackages(this.completionToken, scope, false, false, new ObjectVector());
```

Comment:
as all this is under the flag `isSwitchEnumOrType` we shouldn't reach this line for string switches, but we do.

One indication for a true type switch would be that the resolvedType is not `final`, because then we don't have any subtypes to propose. Trying this in the test, reveals that `jclMin17.jar` contains a stub for `String` that does not declare the class as `final` (should be corrected anyway).

Alternatively, we could explicitly exclude from `isSwitchEnumOrType` if `resolvedType.id == TypeIds.T_javaLangString`.